shared_model/backend/protobuf/queries/proto_query.hpp

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_SHARED_MODEL_PROTO_QUERY_HPP
+#define IROHA_SHARED_MODEL_PROTO_QUERY_HPP
+
+#include <boost/range/numeric.hpp>
+
+#include "backend/protobuf/common_objects/trivial_proto.hpp"
+#include "backend/protobuf/common_objects/signature.hpp"
+#include "backend/protobuf/queries/proto_get_account.hpp"
+#include "interfaces/queries/query.hpp"
+#include "queries.pb.h"
+#include "utils/lazy_initializer.hpp"
+#include "utils/variant_deserializer.hpp"
+
+template <typename... T, typename Archive>
+auto load_query(Archive &&ar) {
+  int which = ar.payload()
+                  .GetDescriptor()
+                  ->FindOneofByName("query")
+                  ->field(ar.payload().query_case())
+                  ->index_in_oneof();
+  return shared_model::detail::variant_impl<T...>::template load<
+      shared_model::interface::Query::QueryVariantType>(
+      std::forward<Archive>(ar), which);
+}
+
+namespace shared_model {
+  namespace proto {
+    class Query final : public CopyableProto<interface::Query,
+                                             iroha::protocol::Query,
+                                             Query> {
+     private:
+      /// polymorphic wrapper type shortcut
+      template <typename Value>
+      using w = detail::PolymorphicWrapper<Value>;
```

Comment:
It's better to use [this](https://github.com/hyperledger/iroha/blob/e42fa486452a7d6ba8def8f39981011236cfc27b/shared_model/backend/protobuf/transaction_responses/proto_tx_response.hpp#L53)